modules/os-eap-datasource/1.0/added/launch/datasource-common.sh

```diff
@@ -307,6 +310,17 @@ function generate_external_datasource() {
            </validation>"
   fi
 
+  if [ -n "$conn_properties" ]; then
+    properties_array=$(echo $conn_properties | tr ';' '\n' )
+
+    for element in $properties_array; do
```

Comment:
If `$properties_array` is to be array, it should be changed to `for element in "${properties_array[@]}"; do`